test/cpp/interop/http2_client.cc

```diff
@@ -134,20 +134,29 @@ bool Http2Client::DoRstDuringData() {
 bool Http2Client::DoGoaway() {
   gpr_log(GPR_DEBUG, "Sending two RPCs and expecting goaway");
 
-  int numCalls = 2;
-  for (int i = 0; i < numCalls; i++) {
-    ClientContext context;
-    SimpleRequest request;
-    SimpleResponse response;
-    request.set_response_size(kLargeResponseSize);
-    grpc::string payload(kLargeRequestSize, '\0');
-    request.mutable_payload()->set_body(payload.c_str(), kLargeRequestSize);
-
-    Status s = serviceStub_.Get()->UnaryCall(&context, request, &response);
-    AssertStatusCode(s, grpc::StatusCode::OK);
-    GPR_ASSERT(response.payload().body() ==
-               grpc::string(kLargeResponseSize, '\0'));
-  }
+  SimpleRequest request;
+  request.set_response_size(kLargeResponseSize);
+  grpc::string payload(kLargeRequestSize, '\0');
+  request.mutable_payload()->set_body(payload.c_str(), kLargeRequestSize);
+
+  ClientContext context1;
+  SimpleResponse response1;
+  Status s = serviceStub_.Get()->UnaryCall(&context1, request, &response1);
+  AssertStatusCode(s, grpc::StatusCode::OK);
+  GPR_ASSERT(response1.payload().body() ==
+             grpc::string(kLargeResponseSize, '\0'));
+
+  // Sleep for one second to give time for client to receive goaway frame.
+  gpr_timespec sleep_time = gpr_time_add(
```

Comment:
`gpr_sleep_until` is used in other C++ tests, e.g., https://github.com/grpc/grpc/blob/9b1ad661bacfc232da2991d24fc2da35fae9285a/test/cpp/end2end/server_crash_test.cc#L77. If the other tests should also be using sleep(1), I can change this, but didn't want to assume it was avoided for no reason - it looks like the only sleep that doesn't use `gpr_sleep_until` is in the route chat example.